File: Data Structures & Algorithms/concatenation-of-array/submission-5.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int length = nums.length;
        int[] ans = new int[length*2];
        for(int i =0; i<length; i++){
            ans[i]=ans[i+length]=nums[i];
        }
        return ans;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-5.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    productExceptSelf(nums) {
        let n = nums.length
        const re = [1]
        for(let i=1; i<n;i++){
            re[i]=re[i-1]*nums[i-1]
        }

        let temp=1;
        for(let i=n-1;i>=0;i--){
            re[i]*=temp;
            temp*=nums[i]
        }
        return re;
    }
}
